tests/ludwig/profiling/test_dataset_profile.py

```diff
@@ -1,11 +1,50 @@
-from ludwig.profiling import dataset_profile_pb2
+import pandas as pd
+import pytest
 
+from ludwig.profiling.dataset_profile import (
+    get_column_profile_summaries_from_proto,
+    get_dataset_profile_proto,
+    get_dataset_profile_view,
+)
+from tests.integration_tests.utils import category_feature, generate_data_as_dataframe, number_feature
 
-def test_dataset_profile_works():
-    dataset_profile = dataset_profile_pb2.DatasetProfile()
-    dataset_profile.num_examples = 10
 
-    from_serialized = dataset_profile_pb2.DatasetProfile()
-    from_serialized.ParseFromString(dataset_profile.SerializeToString())
+def test_get_dataset_profile_view_works():
+    df = pd.DataFrame(
+        {
+            "animal": ["lion", "shark", "cat", "bear", "jellyfish", "kangaroo", "jellyfish", "jellyfish", "fish"],
+            "legs": [4, 0, 4, 4.0, None, 2, None, None, "fins"],
+            "weight": [14.3, 11.8, 4.3, 30.1, 2.0, 120.0, 2.7, 2.2, 1.2],
+        }
+    )
 
-    assert from_serialized.num_examples == 10
+    dataset_profile_view = get_dataset_profile_view(df)
+    dataset_profile_view_proto = get_dataset_profile_proto(dataset_profile_view)
+    column_profile_summaries = get_column_profile_summaries_from_proto(dataset_profile_view_proto)
+
+    assert set(column_profile_summaries.keys()) == {
```

Comment:
Hey @justinxzhao , a question: where do we want to handle `NaN`s and making them strings? Should that only be a concern when converting this data to JSON?

Reply:
Hey @martindavis! 

I imagine that something like generating valid JSON from the whylogs proto for profile visualizations would be handled outside of Ludwig like in the engine, closer to the frontend.